USF 2382 - In cart flavor & size editing

cctina516 opened the pull request:
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix Ticket: https://jira.corp.adobe.com/browse/USF-2382

## Overview

This PR enhances the cart and mini-cart user experience by implementing in-place product editing through a modal interface. The key improvement replaces the previous redirect-to-PDP workflow with a contextual mini-PDP modal that opens directly from cart edit buttons, allowing users to modify product configurations without leaving their current page. The implementation includes comprehensive success messaging that appears in both cart locations simultaneously, auto-dismissing notifications for better UX, and proper internationalization support. Additionally, the PR introduces configuration-based edit button visibility, where main cart shows edit buttons by default while mini-cart requires explicit configuration, ensuring flexible deployment options across different commerce implementations.


## Key Features

### Mini-PDP Modal Integration
- **Mini-cart now opens mini-PDP modal** instead of redirecting to full PDP when clicking "Edit"
- Consistent behavior between main cart and mini-cart edit functionality
- Modal opens with current cart item's selected options and quantity already configured
- Modal includes product configuration options, quantity selection, and price display
- Multiple ways to close modal without making changes (cancel button, close button)
- View all details button redirects users to full PDP page
### Configuration-Based Edit Button Display
- Edit button shows by default in main cart for configurable products
- Edit button does NOT show by default in mini-cart
### Enhanced Success Message System
- **Auto-dismissal**: Success messages automatically disappear after 5 seconds (cart) / 3 seconds (mini-cart)
- **Dual messaging**: Product updates show success messages in both main cart page and mini-cart simultaneously
- **Manual dismiss**: Users can manually close messages before auto-timeout
- **Loading states**: Update button shows "Updating..." text and disabled state during API calls

Test URLs:
- Before: https://main--aem-boilerplate-commerce--hlxsites.aem.live/
- After: https://editing--aem-boilerplate-commerce--hlxsites.aem.live/
https://editing--aem-boilerplate-commerce--hlxsites.aem.live/cart

fnhipster commented on blocks/commerce-cart/commerce-cart.js:
```diff
@@ -94,6 +101,73 @@ export default async function decorate(block) {
+            productName,
+          );
+
+          UI.render(InLineAlert, {
```
To avoid potential side effects on the virtual DOM, it might be better to remove the render entirely instead of clearing the inner HTML of `$notification`.

You can use the returned render:

```js
const inlineAlert = await UI.render(InLineAlert, {...});
```

and remove:

```js
inlineAlert.remove();
```

cctina516 replied:
It's updated, thanks!

fnhipster commented on blocks/commerce-mini-cart/commerce-mini-cart.js:
```diff
@@ -48,6 +61,72 @@ export default async function decorate(block) {
+            '.cart__notification',
+          );
+          if (cartNotification) {
+            UI.render(InLineAlert, {
+              heading: message,
+              type: 'success',
+              variant: 'primary',
+              icon: h(Icon, { source: 'CheckWithCircle' }),
+              'aria-live': 'assertive',
+              role: 'alert',
+              onDismiss: () => {
+                cartNotification.innerHTML = '';
+              },
+            })(cartNotification);
+
+            // Auto-dismiss after 5 seconds
+            setTimeout(() => {
+              cartNotification.innerHTML = '';
+            }, 5000);
+          }
+
+          // Also trigger message in the mini-cart
+          showMessage(message);
+        },
+        () => {
+          if (currentModal) {
+            currentModal.removeModal();
+            currentModal = null;
+          }
+        },
+      );
```
Like in the previous comment, could you consider removing the element from the DOM instead of clearing its inner HTML to avoid potential virtual DOM side effects?

fnhipster reviewed (changes requested):
Good stuff! This will help streamline the use of "Mini PDP" in modals across other areas, like "Add to Cart" from Product Lists, PREX, etc.

ericeoeur commented:
@cctina516 Excellent work, Christina! I just have styling consistencies that needs to be addressed: styling is inconsistent between the mini-cart's MINI PDP Modal and the regular cart's Mini PDP Modal. 

Buttons on the bottom for mini-cart's mini pdp modal is missing styling in the `.mini-pdp__buttons` class, which is causing the buttons to not have any margin between line breaks and when in mobile view, the buttons are not at full width as they are in the cart's version of the mini pdp modal. Additionally, title and price (header) are not consistent between the mini-cart's modal and the cart's modal. Based off the [figma designs](https://www.figma.com/design/xtD0uYcFdRQxUnL3cXumXQ/V2-Base-Design-System?node-id=12069-69943&t=Ee0xqPO3fcxT0qTh-0) it seems like they should be consistent with one another. 

<img width="1886" height="1117" alt="image" src="https://github.com/user-attachments/assets/25b7338e-5d10-4cdb-b3dc-546451077ef0" />

cctina516 commented:
> @cctina516 Excellent work, Christina! I just have styling consistencies that needs to be addressed: styling is inconsistent between the mini-cart's MINI PDP Modal and the regular cart's Mini PDP Modal.
> 
@ericeoeur Thanks Eric, good catch. I moved the styling to the mini-pdp.css, this should be fixed now. 
> Also, the cypress test that is failing seems unrelated (wishlist) but please look into that to confirm.

yea it's not related. This is now fixed by merging in the latest from integration.
